src/hip_module.cpp

```diff
@@ -565,6 +600,19 @@ hipError_t hipLaunchCooperativeKernelMultiDevice(hipLaunchParams* launchParamsLi
             }
             return ihipLogStatus(hipErrorLaunchFailure);
         }
+        //initialize and setup the implicit kernel argument for multi-grid sync
+        mg_info_ptr[i]->mgs       = mg_sync_ptr;
+        mg_info_ptr[i]->grid_id   = i;
+        mg_info_ptr[i]->num_grids = numDevices;
+        mg_info_ptr[i]->all_sum   = all_sum;
+        mg_info_ptr[i]->prev_sum  = 0;
+        for (int j = 0; j < i; ++j) {
```

Comment:
Wouldn't it be easier and faster to keep a running sum in the i loop instead of recalculating prev_sum on each iteration?

Reply:
Agree. Modified code as requested.